### ISSUE TITLE
Improve test matrix and build tools

### ISSUE THREAD

**Issue by zerolab** [torchbox/wagtail-grapple#279]
- [ ] move to tox for greater control e.g. test against python 3.11 (but exclude Wagtail 2.15)
- [ ] move to flit for build
- [ ] consolidate dev/test requirements

**Comment by dopry** [comment]
should we also consider implementing modernized packaging with [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)? With metadata per [PEP 621](https://peps.python.org/pep-0621/) we can specify grouped dependencies. I'm not sure how it would impact the rest of the build and release pipeline. I've been slowly converting all of my personal project to use pyproject.toml with poetry in place of pip for dependency management.

**Comment by zerolab** [comment]
yes to pyproject.toml, but flit for the build/release. IMHO poetry is too bloated for this.
https://github.com/wagtail/cookiecutter-wagtail-package/pull/32 for reference

**Comment by dopry** [comment]
I only use poetry as a package manager, not a build tool...  I continue to use setuptools. I'll have to check flit out after eoy.

**Comment by zerolab** [comment]
I found the whole package/publish to PyPi experience so much nicer (and faster) with flit.
However, it doesn't do build steps, so setuptools or others are better for that

**Comment by dopry** [comment]
I don't think I need build steps. I pretty much only work on pure python projects... Do that extends to deps that need build steps, like pillow, psycopg, etc?
